include/xtl/xparallel_foreach.hpp

```diff
@@ -0,0 +1,349 @@
+#ifndef XTL_XPARALLEL_FOR_EACH_HPP
+#define XTL_XPARALLEL_FOR_EACH_HPP
+
+#define XTL_DEFAULT_CHUNKING_FACTOR 3.0
+
+#include <future>
+#include <type_traits>
+
+#include "xtl/xthreadpool.hpp"
+
+namespace xtl
+{      
+    ///\cond
+    namespace detail{
+        template<class INTEGER, class F,typename std::enable_if<std::is_integral<INTEGER>::value ,int>::type = 0>
+        void xparallel_integer_foreach_impl(
+            const INTEGER size,
+            xthreadpool & pool,
+            F && f
+        ){
```

Comment:
Formatting:
- upper case names are reserved for macros
- no space between a type name and '&' or '&&'

I don't understand why you use SFINAE here since the method is not overloaded / specialized elsewhere.